### PR TITLE
[metricbeat] Migrate nginx/substatus to reporterv2 with error

### PR DESCRIPTION
**Pull request by fearful-symmetry**
See  #11374 

There was a community PR for this, but it's been abandoned so I'm just doing it myself. This one is a bit odd, since it requires changes to some other metricbeat tests, which will look for error strings.

### CODE REVIEW

File: metricbeat/module/nginx/stubstatus/stubstatus_integration_test.go [hunk 1]
```diff
@@ -48,8 +48,8 @@ func TestFetch(t *testing.T) {
 func TestData(t *testing.T) {
 	service := compose.EnsureUp(t, "nginx")
 
-	f := mbtest.NewReportingMetricSetV2(t, getConfig(service.Host()))
-	if err := mbtest.WriteEventsReporterV2(f, t, ""); err != nil {
```
Comment (jsoriano) [inline metricbeat/module/nginx/stubstatus/stubstatus_integration_test.go]: Or even better :slightly_smiling_face: 
```
f := mbtest.NewFetcher((t, getConfig(service.Host())))
f.WriteEvents(t, "")
```

**Comment by fearful-symmetry** [comment]
jenkins, test this

**Comment by fearful-symmetry** [comment]
I don't think the CI failures I'm seeing are related.

**Comment by fearful-symmetry** [comment]
jenkins, test this

**Pull request merged**